### PR TITLE
New email authentification method

### PR DESCRIPTION
**Pull request by jitka1997**

### CODE REVIEW

File: lib/components/login.tsx [hunk 1]
```diff
@@ -1,18 +1,111 @@
...
+  }
+
+  return (
+    <>
+      <Dialog open={open} onClose={handleClose}>
+        <DialogTitle>{action}</DialogTitle>
+        <DialogContent>
+          <TextField
+            autoFocus
+            margin="dense"
```
[the remark this answers is not in the document]
Comment (jitka1997) [reply lib/components/login.tsx:66]: I think it looks better with margin dense, the textfields aren't so close to each other

File: lib/components/login.tsx [hunk 2]
```diff
@@ -1,18 +1,113 @@
...
 import { LoadingScreen } from './loading'
 
+const useStyles = makeStyles((theme: Theme) => ({
+  signInButton: {
+    display: 'flex',
+    flexDirection: 'column',
+    width: 240,
+    margin: 'auto',
+  },
+  or: {
```
Comment (Siegrift) [inline lib/components/login.tsx:30]: Nit: `orText`

File: lib/components/login.tsx [hunk 3]
```diff
@@ -1,18 +1,113 @@
...
+  const [action, setAction] = useState('Sign in')
+  const dispatch = useDispatch()
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+
+  const handleClose = () => {
+    setOpen(false)
+    setAction('Sign in')
+  }
+  const handleAction = async () => {
```
Comment (Siegrift) [inline lib/components/login.tsx:53]: Does this handle errors as well?

File: lib/components/login.tsx [hunk 4]
```diff
@@ -1,18 +1,113 @@
...
+            label="Password"
+            type="password"
+            fullWidth
+            value={password}
+            onChange={(e) => setPassword(e.target.value)}
+          />
+          <Link
+            component="button"
+            variant="body2"
+            onClick={() => setAction(action === 'Sign in' ? 'Sign up' : 'Sign in')}
```
Comment (Siegrift) [inline lib/components/login.tsx:88]: Pressing enter should trigger sign in.

File: lib/components/login.tsx [hunk 5]
```diff
@@ -1,18 +1,113 @@
...
+          >
+            {action === 'Sign in' ? 'or create new account' : 'or sign in with an existing account'}
+          </Link>
+        </DialogContent>
+        <DialogActions>
+          <Button onClick={handleClose} color="primary">
+            Cancel
+          </Button>
+          <Button onClick={handleAction} color="primary">
```
Comment (Siegrift) [inline lib/components/login.tsx:97]: Pressing enter in the form fields should also trigger sign in

File: lib/components/login.tsx [hunk 6]
```diff
@@ -1,18 +1,113 @@
...
+  const [password, setPassword] = useState('')
+
+  const handleClose = () => {
+    setOpen(false)
+    setAction('Sign in')
+  }
+  const handleAction = async () => {
+    if (action === 'Sign in') {
+      await dispatch(authChangeAction('loggingIn', null))
+      signInWithEmailAndPassword(email, password)
```
Comment (Siegrift) [inline lib/components/login.tsx:56]: This returns a promise so you need to `await` it.

**Review by Siegrift — APPROVED** [review]

**Pull request merged**